### ISSUE TITLE
Feature: Expose alz provider arguments for Private DNS and Policy Assignment addition

### ISSUE THREAD

**Issue by OmnipotentOwl** [Azure/terraform-azurerm-avm-ptn-alz#10]
## Summary
In testing the module and using it for deployments I found that there are certain configuration properties available in the alz provider that are not exposed in the module to allow for configuration in detail of an Enterprise Scale platform landing zone setup.

In specific the alz provider exposes the [policy_assignments_to_add](https://registry.terraform.io/providers/Azure/alz/latest/docs/data-sources/archetype#nested-schema-for-policy_assignments_to_add) and [private_dns_zone_resource_group_id](https://registry.terraform.io/providers/Azure/alz/latest/docs/data-sources/archetype#private_dns_zone_resource_group_id) arguments that allow for configuration of Policy Assignments using the provider to override library default values with environment specific values.

Below I have included an example of the implementation proposed for the module based on what the provider documentation offers and can submit this as a PR.

## Examples
```hcl
variable "policy_assignments_to_add" {
  type = map(object({
    display_name               = optional(string, null)
    enforcement_mode           = optional(string, null)
    identity                   = optional(string, null)
    identity_ids               = optional(list(string), null)
    policy_definition_id       = optional(string, null)
    policy_definition_name     = optional(string, null)
    policy_set_definition_name = optional(string, null)
    parameters                 = optional(string, null)
  }))
  default     = {}
  description = <<DESCRIPTION
A map of policy assignment objects to add or update the alz archetype with.
DESCRIPTION
}
```
Would replace the existing definition by including the object optional arguments per the provider spec. For the `private_dns_zone_resource_group_id` this could be implemented using the prior art for the `default_log_analytics_workspace_id` variable as a reference.

```hcl
variable "default_private_dns_zone_resource_group_id" {
  type        = string
  description = <<DESCRIPTION
DESCRIPTION
  default     = null
}
```

This would then result in an update to the `data.alz_archetype.this` definition to include in the `defaults` block the `private_dns_zone_resource_group_id` variable reference.

```hcl
data "alz_archetype" "this" {
  id = var.id
  defaults = {
    location                           = var.default_location
    log_analytics_workspace_id         = var.default_log_analytics_workspace_id
    private_dns_zone_resource_group_id = var.default_private_dns_zone_resource_group_id
  }
...
```

**Comment by matt-FFFFFF** [comment]
Hi @OmnipotentOwl 

Thanks for this, the feedback is much appreciated. I am back in the office next week and will respond properly then.

**Comment by matt-FFFFFF** [comment]
fixed in #11
